From 0d4de74341dfd2b90b09bc39fa1cff2ff3c26866 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E2=80=9Chuangyifeng=E2=80=9D?=
 <huangyifeng@eswincomputing.com>
Date: Fri, 10 May 2024 16:02:53 +0800
Subject: [PATCH 018/109] WIN2030-15099:refactor(bmc):bmc web cmd support

Changelogs: 1.remove uart4_init from board_init
Change-Id: I4dfe30599c3f99742f8f790a8709e18dd7505490
---
 Core/Src/hf_board_init.c       | 7 +++++--
 Core/Src/hf_protocol_process.c | 3 ---
 2 files changed, 5 insertions(+), 5 deletions(-)

diff --git a/Core/Src/hf_board_init.c b/Core/Src/hf_board_init.c
index f61da93..c144f67 100644
--- a/Core/Src/hf_board_init.c
+++ b/Core/Src/hf_board_init.c
@@ -102,8 +102,12 @@ void i2c_deinit(I2C_TypeDef *Instance)
 
 void uart_init(USART_TypeDef *Instance)
 {
-	if (Instance == UART4)
+	if (Instance == UART4) {
 		MX_UART4_Init();
+		//trigger uart rx
+		HAL_UARTEx_ReceiveToIdle_DMA(&huart4, (int8_t *)&UART4_RxMsg,
+			sizeof(UART4_RxMsg));
+	}
 	else if (Instance == USART6)
 		MX_USART6_UART_Init();
 	else
@@ -134,7 +138,6 @@ int board_init(void)
 	MX_DMA_Init();
 	MX_SPI2_Init();
 	MX_USART3_UART_Init();
-	MX_UART4_Init();
 	MX_RTC_Init();
 	MX_TIM4_Init();
 	MX_SPI1_Init();
diff --git a/Core/Src/hf_protocol_process.c b/Core/Src/hf_protocol_process.c
index c105c4c..1d9aed4 100644
--- a/Core/Src/hf_protocol_process.c
+++ b/Core/Src/hf_protocol_process.c
@@ -701,9 +701,6 @@ void uart4_protocol_task(void *argument)
 
 	//Init web server cmd list
 	vListInitialise(&WebCmdList);
-
-	//trigger uart rx
-	HAL_UARTEx_ReceiveToIdle_DMA(&huart4, (int8_t *)&UART4_RxMsg, sizeof(UART4_RxMsg));
 	for (;;) {
 		if (xQueueReceive(xUart4MsgQueue, &(msg), portMAX_DELAY)) {
 			if (msg.header == FRAME_HEADER && msg.tail == FRAME_TAIL) {
-- 
2.25.1

